lib/derivations.nix

```diff
@@ -206,4 +206,40 @@ in
   optionalDrvAttr =
     cond:
     value: if cond then value else null;
+
+  /**
+    Wrap a derivation such that instantiating it produces a warning.
+
+    # Inputs
+
+    `msg`
+    : The warning message to emit (via lib.warn).
+
+    `pkg`
+    : The derivation to wrap.
+
+    # Examples
+    :::{.example}
+    ## `lib.derivations.wrapWithWarn` usage example
+
+    ```nix
+    {
+      myPackage = wrapWithWarn "myPackage has been renamed to my-package" my-package;
+    }
+    ```
+
+    :::
+  */
+  wrapWithWarn =
```

Comment:
Inherit it in `lib/default.nix` and it'll be available in `lib`

Reply:
Thanks! I wonder how that worked locally.